File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/team/lobby/starter/StarterLobbyListener.java
package com.gmail.borlandlp.minigamesdtools.arena.team.lobby.starter;

import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventHandler;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventPriority;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerLeaveLocalEvent;
import com.gmail.borlandlp.minigamesdtools.arena.team.lobby.ArenaLobby;

public class StarterLobbyListener implements ArenaEventListener {
    private StarterLobby lobby;

    public StarterLobbyListener(StarterLobby l) {
        this.lobby = l;
    }

    @ArenaEventHandler(priority = ArenaEventPriority.HIGHEST)
    public void onPJoin(ArenaPlayerJoinLocalEvent event) {
        if(((ArenaLobby) this.lobby).isEnabled()) {
            this.lobby.addPlayer(event.getPlayer());
        }
    }

    @ArenaEventHandler(priority = ArenaEventPriority.HIGHEST)
    public void onPLeave(ArenaPlayerLeaveLocalEvent event) {
        if(((ArenaLobby) this.lobby).isEnabled()) {
            this.lobby.removePlayer(event.getPlayer());
        }
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/team/lobby/starter/StarterLobbyListener.kt
package com.gmail.borlandlp.minigamesdtools.arena.team.lobby.starter

import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventHandler
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaEventPriority
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerJoinLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.localevent.ArenaPlayerLeaveLocalEvent
import com.gmail.borlandlp.minigamesdtools.arena.team.lobby.ArenaLobby

class StarterLobbyListener(private val lobby: StarterLobby) : ArenaEventListener {
    @ArenaEventHandler(priority = ArenaEventPriority.HIGHEST)
    fun onPJoin(event: ArenaPlayerJoinLocalEvent) {
        if ((lobby as ArenaLobby).isEnabled) {
            lobby.addPlayer(event.player)
        }
    }

    @ArenaEventHandler(priority = ArenaEventPriority.HIGHEST)
    fun onPLeave(event: ArenaPlayerLeaveLocalEvent) {
        if ((lobby as ArenaLobby).isEnabled) {
            lobby.removePlayer(event.player)
        }
    }

}